pkg/kv/kvserver/rangefeed/scheduled_processor.go

```diff
@@ -304,21 +320,31 @@ func (p *ScheduledProcessor) Register(
 	withOmitRemote bool,
 	stream Stream,
 	disconnectFn func(),
-) (bool, *Filter) {
+) (bool, Disconnector, *Filter) {
 	// Synchronize the event channel so that this registration doesn't see any
 	// events that were consumed before this registration was called. Instead,
 	// it should see these events during its catch up scan.
 	p.syncEventC()
 
 	blockWhenFull := p.Config.EventChanTimeout == 0 // for testing
+
 	var r registration
-	if _, ok := stream.(BufferedStream); ok {
-		log.Fatalf(context.Background(),
-			"unimplemented: unbuffered registrations for rangefeed, see #126560")
+	bufferedStream, isBufferedStream := stream.(BufferedStream)
+	if isBufferedStream {
+		r = newUnbufferedRegistration(streamCtx,
+			span.AsRawSpanWithNoLocals(), startTS, catchUpIter, withDiff, withFiltering, withOmitRemote,
+			p.Config.EventChanCap, p.Metrics, bufferedStream, disconnectFn,
+			func(ctx context.Context, r registration) {
+				p.unregisterClientAsync(r)
+			})
 	} else {
 		r = newBufferedRegistration(
+			streamCtx,
 			span.AsRawSpanWithNoLocals(), startTS, catchUpIter, withDiff, withFiltering, withOmitRemote,
 			p.Config.EventChanCap, blockWhenFull, p.Metrics, stream, disconnectFn,
+			func(ctx context.Context, r registration) {
+				p.unregisterClientAsync(r)
+			},
 		)
 	}
```

Comment:
Let's not change it now, but as a potential follow up, a way to do this without the type switch would be for stream to have a method like SendIsNonBlocking().

Reply:
I have some bad news: we can’t completely remove BufferedStream as we had done this morning because we still rely on the two different structs to determine which type of rangefeed registration to create right here. What you proposed here would help us solve this. 

But I’m also not sure whether adding a method like SendIsNonBlocking to the Stream interface is a good idea. Stream interface is used across so many test implementations. And it seems nicer to keep this interface minimal. Perhaps we could discuss this further and address this as a follow up.